packages/react/src/components/Button/index.tsx

```diff
@@ -1,42 +1,40 @@
-import React from 'react';
-import PropTypes from 'prop-types';
+import React, { ButtonHTMLAttributes, forwardRef, ReactNode } from 'react';
 import classNames from 'classnames';
 
-export interface ButtonProps
-  extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+export interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
   variant?: 'primary' | 'secondary' | 'error' | 'link';
-  buttonRef?: React.Ref<HTMLButtonElement>;
-  disabled?: boolean; // todo, investigate why disabled is needed
   thin?: boolean;
+  startIcon?: ReactNode;
+  endIcon?: ReactNode;
 }
 
-export default class Button extends React.Component<ButtonProps> {
-  static defaultProps = {
-    variant: 'primary',
-    buttonRef: () => {}
-  };
-
-  static propTypes = {
-    variant: PropTypes.oneOf(['primary', 'secondary', 'error', 'link']),
-    children: PropTypes.node,
-    className: PropTypes.string,
-    buttonRef: PropTypes.oneOfType([
-      PropTypes.func,
-      PropTypes.shape({ current: PropTypes.any })
-    ])
-  };
-
-  static displayName = 'Button';
-
-  render() {
-    const {
-      variant,
+const Button = forwardRef<HTMLButtonElement, ButtonProps>(
```

Comment:
I'm in favor of using `forwardRef` here, but this is a breaking change. I think we should eventually change this, but I don't think it should be a part of this PR.